systemtest/src/main/java/io/strimzi/systemtest/utils/FileUtils.java

```diff
@@ -30,132 +30,73 @@ private FileUtils() { }
 
     @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_BAD_PRACTICE")
     public static File downloadAndUnzip(String url) {
-        File dir = null;
-        FileOutputStream fout = null;
-        ZipInputStream zin = null;
-        try {
-            InputStream bais = (InputStream) URI.create(url).toURL().openConnection().getContent();
-            dir = Files.createTempDirectory(FileUtils.class.getName()).toFile();
+
+        try (InputStream bais = (InputStream) URI.create(url).toURL().openConnection().getContent();
+             ZipInputStream zin = new ZipInputStream(bais)){
+
+            File dir = Files.createTempDirectory(FileUtils.class.getName()).toFile();
             dir.deleteOnExit();
-            zin = new ZipInputStream(bais);
+
             ZipEntry entry = zin.getNextEntry();
             byte[] buffer = new byte[8 * 1024];
             int len;
             while (entry != null) {
                 File file = new File(dir, entry.getName());
-                if (entry.isDirectory()) {
```

Comment:
This case of the zip entry being a directory is no longer handled.

Reply:
So you propose to make something like that 
```
               if (entry.isDirectory()) {
                    LOGGER.info("Entry is not directory...?");
                    file.mkdirs();
                } else {
                    try (FileOutputStream fout = new FileOutputStream(file)) {
                        while ((len = zin.read(buffer)) != -1) {
                            fout.write(buffer, 0, len);
                        }
                    }
                }
```
What actually `file.mkdirs();` will do?

Reply:
There's this thing called Javadoc, you really should check it out :wink:... It will create directories which don't exist. `mkdir()` (singular) will just create a single dir, failing if the parent doesn't exist. `mkdirs()` will create parent (and all the way up, as necessary).

The truth is that malformed zips are not unheard of (e.g. missing directory entries, or directory entries after file entries in that directory), so you probably should `mkdirs(file.getParentFile())` even in the `else` block.

Reply:
Thanks for the exlanation and sorry for stupid question...Anyway i come up with the following sollution on this problem based on `https://stackoverflow.com/questions/27904329/warning-file-mkdir-is-ignored`